### ISSUE TITLE
[File a bug report for the Demon Implant]: Do not work x32 x86 bit payloads

### ISSUE THREAD

**Issue by SergANT85** [HavocFramework/Havoc#398]
### What happened?

Hello!

I generate format x32 x86 bit payload, and it do not work. But when I am testing on x64 platform - it work good.

Also I have problem inject bin file x32 format. 

X64 is working very well.

Tnx.

### Did You Do a Pull First?

Latest (You performed a pull first)

### Did You Try With the Dev Branch?

Yes (You tried using the dev branch but the problem persist)

### Did You Read Over Your Issue First?

- [X] I declare I made an effort and provided the necessary information for replication of the issue.

**Comment by curie71** [comment]
I found the same problem!
<img width="726" alt="033d782380bacaa4e3aab37cae84b45" src="https://github.com/HavocFramework/Havoc/assets/39853223/33e06094-3296-4a8a-b43b-3d6d7948022a">

**Comment by muhyuddin** [comment]
Facing same issue, getting Access Violation error in Windows 7 32 bit, when running x86 executable. Whille running in WinDbg ret address goes to 00000000, while reading out of bound data. I might be wrong.

**Comment by curie71** [comment]
> Facing same issue, getting Access Violation error in Windows 7 32 bit, when running x86 executable. Whille running in WinDbg ret address goes to 00000000, while reading out of bound data. I might be wrong.

It seems like the newest code solve the problem.

**Comment by muhyuddin** [comment]
Nope, crashing on the latest code. I just cloned and build and issue is same. 
Windows 7 Ultimate
Version 6.1 (Build 7601: Service pack 1)

**Comment by muhyuddin** [comment]
Any updates ? @SergANT85 

I have debugged the shellcode, its crashing on NtProtectVirtualMemory call inside the shellcode. Cant traceback the issue in C code. Works on WoW64, but crashes on x86 with ACCESS_VIOLATION.

**Comment by MrEmpy** [comment]
The same problem happens to me. When booting demon x86 I just see the process die quickly. I'm using Windows 7 Enterprise Service Pack 1
